files/image_config/platform/rc.local

```diff
@@ -29,8 +39,60 @@ if [ ! -e /host/machine.conf ]; then
         done
     fi
 
+    # Extract the previous NOS's partition that contains the migration artifacts
+    set -- $(cat /proc/cmdline)
+    for x in "$@"; do
+        case "$x" in
+            nos-config-part=*)
+                nos_val="${x#nos-config-part=}"
+                ;;
+        esac
+    done
+
+    if [ -n "$nos_val" ]; then
+        nos_dev=$(findfs $nos_val)
+        if [ $? != 0 ]; then
+            log_migration "ERROR: nos_dev not found. Check grub parameters"
+        fi
+    else
+        log_migration "ERROR: nos_val not found. Check grub parameters"
+    fi
+
+    if [ -n "$nos_dev" ]; then
+        # Mount the previous NOS's partition
+        mkdir -p /mnt/nos_migration
+        mount $nos_dev /mnt/nos_migration
+        mkdir -p /etc/sonic/migration
+
+        # Copy relevant files
+        nos_migration_import /mnt/nos_migration/minigraph.xml /etc/sonic/migration
+        nos_migration_import /mnt/nos_migration/arp.json /etc/sonic/migration
+        nos_migration_import /mnt/nos_migration/fdb.json /etc/sonic/migration
+        nos_migration_import /mnt/nos_migration/mgmt_interface.mac /etc/sonic/migration
+
+        umount /mnt/nos_migration
+        rmdir /mnt/nos_migration
+    fi
+
+    # Extract, validate and set the eth0's mac address
+    if [ -f /etc/sonic/migration/mgmt_interface.mac ]; then
+       eth0_mac=$(grep "macaddr" /etc/sonic/migration/mgmt_interface.mac | awk -F'=' '{print $2}')
+       log_migration "Setting eth0 mac as $eth0_mac."
+       if [ `echo $eth0_mac | egrep "^([0-9a-fA-F]{2}:){5}[0-9a-fA-F]{2}$"` ]; then
+           ifconfig eth0 down
+           ip link set eth0 address $eth0_mac
+           ifconfig eth0 up
+           rm /etc/sonic/migration/mgmt_interface.mac
+       else
+           log_migration "ERROR: $eth0_mac: invalid MAC!"
+       fi
+    fi
+
     migration="TRUE"
     umount /mnt/onie-boot
+else
+    # this is a reboot post NOS migration : remove stale imported files that were used during migration.
+    rm -fr /etc/sonic/migration
```

Comment:
Alternatively, we could remove the arp.json & fdb.json after checking that orchagent initialized but removing here localizes the migration changes to rc.local.